gps-nmea/nmea_parser.go

```diff
@@ -0,0 +1,137 @@
+package nmea
+
+import (
+	"math"
+	"strconv"
+	"strings"
+
+	"github.com/adrianmo/go-nmea"
+	geo "github.com/kellydunn/golang-geo"
+	"github.com/pkg/errors"
+	"go.uber.org/multierr"
+)
+
+const (
+	knotsToMPerSec = 0.51444
+	kphToMPerSec   = 0.27778
+)
+
+type gpsData struct {
+	location   *geo.Point
+	alt        float64
+	speed      float64 // ground speed in m per sec
+	vDOP       float64 // vertical accuracy
+	hDOP       float64 // horizontal accuracy
+	satsInView int     // quantity satellites in view
+	satsInUse  int     // quantity satellites in view
+	valid      bool
+	fixQuality int
+}
+
+func errInvalidFix(sentenceType, badFix, goodFix string) error {
+	return errors.Errorf("type %q sentence fix is not valid have: %q  want %q", sentenceType, badFix, goodFix)
+}
+
+// parseAndUpdate will attempt to parse a line to an NMEA sentence, and if valid, will try to update the given struct
+// with the values for that line. Nothing will be updated if there is not a valid gps fix.
+func (g *gpsData) parseAndUpdate(line string) error {
```

Comment:
I just realized how crazy this function is. It would be nice if we can break it up into many parts.

Reply:
not here, in rdk, and as a separate ticket so we can export it to this driver eventually.

Reply:
Will make a ticket for this